include/dxc/DxilContainer/DxilPipelineStateValidation.h

```diff
@@ -635,7 +637,8 @@ class DxilPipelineStateValidation {
   _T *GetRecord(void *pRecords, uint32_t recordSize, uint32_t numRecords,
                 uint32_t index) const {
     if (pRecords && index < numRecords && sizeof(_T) <= recordSize) {
-      assert((size_t)index * (size_t)recordSize <= UINT_MAX);
+      if (!((size_t)index * (size_t)recordSize <= UINT_MAX))
```

Comment:
This would always be false right? the product of two uints will always be at most UINT_MAX, otherwise there would be an overflow, so how could it ever exceed UINT_MAX?

Reply:
size_t is 64bits for x64.
UINT_MAX is 32bits.
x86 will always be false.